eos-updater/poll-common.c

```diff
@@ -580,6 +585,24 @@ should_follow_checkpoint (OstreeSysroot     *sysroot,
       return FALSE;
     }
 
+  /* https://phabricator.endlessm.com/T33311 */
+  if (is_conditional_upgrade_path &&
+      sys_vendor != NULL && product_name != NULL &&
+      booted_system_is_unsupported_by_eos4_kernel (sys_vendor, product_name))
+    {
+      *out_reason = g_strdup_printf (_("%s %s systems are not supported in EOS 5."), sys_vendor, product_name);
+      return FALSE;
+    }
```

Comment:
I think this is a no-op since it's exactly the same code as the T31772 block above. I think either you don't need this block or you should add a new `booted_system_is_unsupported_by_eos5_kernel` function and add that to the above conditional. Basically that block would become a "you can no longer upgrade to newer Endless releases for this hardware" conditional.

Reply:
Oops!  Thanks for the catch up!